src/sync/sync_user.cpp

```diff
@@ -325,6 +328,18 @@ std::string SyncUser::access_token() const
     return m_access_token.token;
 }
 
+std::string SyncUser::device_id() const
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    return m_device_id;
+}
+
+bool SyncUser::has_device_id() const
+{
+    std::lock_guard<std::mutex> lock(m_mutex);
+    return !m_device_id.empty() && m_device_id != "000000000000000000000000";
+}
+
```

Comment:
Should we be providing a way to update the device id as well, or is that missing by design so that people can't update it once a SyncUser has been constructed?

Reply:
This is by design, correct.